File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/adapter/ForeignCenterPlayer.java
package jv.com.tencent.cain.design_pattern.adapter;

/**
 * @author cainjiang
 * @date 2020/4/15
 */
public class ForeignCenterPlayer {
    private String name;

    public ForeignCenterPlayer(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void jingong() {
        System.out.println("ForeignCenterPlayer# " + name + " 进攻 ---");
    }

    public void fangshou() {
        System.out.println("ForeignCenterPlayer# " + name + " 防守 ---");
    }

}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/adapter/ForeignCenterPlayer.kt
package kt.com.tencent.cain.desigin_pattern.adapter

/**
 * @author cainjiang
 * @date 2020/4/15
 */
class ForeignCenterPlayer(val name: String) {

    fun jingong() {
        println("ForeignCenterPlayer# $name 进攻")
    }

    fun fangshou() {
        println("ForeignCenterPlayer# $name 防守")
    }
}